src/Microsoft.ML.Core/Utilities/Hashing.cs

```diff
@@ -174,6 +174,94 @@ public static uint MurmurHash(uint hash, ReadOnlySpan<char> span, bool toUpper =
             return hash;
         }
 
+        // MurmurHashV2 is an implementation of MurmurHash3_x86_32 (https://github.com/aappleby/smhasher/blob/master/src/MurmurHash3.cpp#L94) used by Onnxruntime's
+        // MurmurHash operator, implemented to have matching hashing results between ORT and ML.NET.
+        // One key difference between the two implementations is that strings use a different hashing algorithm in the previous version, but
+        // every type uses the same implementation on V2.
+        // The V1 String Hashing Algorithm had the following properities:
+        //     - Case Conversion: used inside the hashing algorithm in ML.Net.
+        //     - Mock UTF8 encoding: strings in C# are UTF16 and need to be converted to UTF8 before hashing
```

Comment:
Please elaborate on what Mock UTF8 encoding is. If I recall correctly, we are omitting certain code pages. 
@KsenijaS Didn't you have an implementation that supported the full UTF8 encoding? Can we use that?

Reply:
Yes, but also partial UTF8 support is sufficient. Mock UTF8 doesn't cover emojis and some special math characters